src/main/java/org/prebid/server/currency/CurrencyConversionService.java

```diff
@@ -137,15 +138,21 @@ public BigDecimal convertCurrency(BigDecimal price, Map<String, Map<String, BigD
             return price;
         }
 
-        // get conversion rate from request currency rates if it is present
         BigDecimal conversionRate = null;
-        if (requestCurrencyRates != null) {
-            conversionRate = getConversionRate(requestCurrencyRates, adServerCurrency, effectiveBidCurrency);
-        }
-
-        // if conversion rate from requestCurrency was not found, try the same from latest currencies
-        if (conversionRate == null) {
-            conversionRate = getConversionRate(externalCurrencyRates, adServerCurrency, effectiveBidCurrency);
+        if (usepbsrates) {
+            conversionRate = getConversionRateFromServer(externalCurrencyRates, adServerCurrency, effectiveBidCurrency,
+                    conversionRate);
+            if (conversionRate == null) {
+                conversionRate = getConversionRateFromRequest(requestCurrencyRates, adServerCurrency,
+                        effectiveBidCurrency, conversionRate);
+            }
```

Comment:
Does this logic can be in the **getConversionRateFromServer** method. 
(First we ask server and then request) 

P.s I do not like multiole **if** so close and another one in the method